cfme/tests/test_rest_no_provider.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+
+from cfme import test_requirements
+from cfme.utils.wait import wait_for
+
+pytestmark = [test_requirements.rest, pytest.mark.tier(1)]
+
+
+@pytest.fixture
+def change_company_name(appliance):
+    old_company_name = appliance.advanced_settings["server"]["company"]
+    updated_company_name = "REST Company Name"
+    appliance.server.settings.update_basic_information({"company_name": updated_company_name})
```

Comment:
Why here do we `update_basic_information` but down below we `update_advanced_settings`? Do they end up being equivalent?

Reply:
No, they are different.
If you look at the bug, updating server roles via REST resets the `Company name` to its default value.
Hence, we first change the company name, which happens via `update_basic_information`, and then use `update_advanced_settings` to update server role via REST.

Reply:
Spoke offline, now using `update_advanced_settings` in teardown and setup.